File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.cs
public class Solution {
    public int WaysToSplitArray(int[] nums) {
        var ret = 0;
        long sum=0, currSum=0;        
        
        foreach(var n in nums)
        {
            sum += n;
        }        
        
        for(int i=0; i < nums.Length-1; i++)
        {
            currSum+=nums[i];
            if(currSum >= sum-currSum) ret++;
        }
        
        return ret;
    }
}

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.ts
function waysToSplitArray(nums: number[]): number {
    let sum:number=0;
    const prefix = nums.map(val => sum+=val);
    
    let ret: number=0;
    
    for(let i=0; i < prefix.length-1; i++){
        if(prefix[i] >= (prefix[prefix.length-1]-prefix[i]))
            ret++;
    }

    return ret;
    
};